sycl/test/basic_tests/known_identity.cpp

```diff
@@ -93,6 +95,288 @@ template <typename T> void checkBoolKnownIdentity() {
   static_assert(known_identity<sycl::logical_or<T>, T>::value == false);
 }
 
+template <typename T, int Num>
+bool compareVectors(const vec<T, Num> a, const vec<T, Num> b) {
+  bool res = true;
+  for (int i = 0; i < Num; ++i) {
+    res &= (a[i] == b[i]);
+  }
+  return res;
+}
+
+template <typename T, int Num>
+typename std::enable_if<!std::is_same<T, half>::value &&
+                            !std::is_same<T, float>::value &&
+                            !std::is_same<T, double>::value,
+                        void>::type
```

Comment:
```suggestion
std::enable_if_t<!std::is_same_v<T, half> && !std::is_same_v<T, float> &&
                 !std::is_same_v<T, double>>
```
Minor stylistic suggestion. Now that we have C++17 for this test and we have `std::enable_if_t` in use in the rest of the changes, I like the brevity of the above code suggestion. 😄

Reply:
The same changes can be applied to the other tests.

Reply:
Done

Reply:
Thanks!